mmv1/products/gkeonprem/BareMetalCluster.yaml

```diff
@@ -0,0 +1,997 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: "BareMetalCluster"
+min_version: beta
+base_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters"
+create_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters?bare_metal_cluster_id={{name}}"
+update_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"
+delete_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}?force=true"
+self_link: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"
+update_verb: :PATCH
+update_mask: true
+description: "A Google Bare Metal User Cluster."
+autogen_async: false
+id_format: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"
+import_format: ["projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"]
+taint_resource_on_failed_create: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_bare_metal_cluster_basic"
+    min_version: beta
+    primary_resource_id: "cluster1"
+    vars:
+      name: "basic"
+    test_env_vars:
+      project: :PROJECT_NAME
+parameters:
+  - !ruby/object:Api::Type::String
+    name: "name"
+    description: The bare metal cluster name.
+    immutable: true
+    url_param_only: true
+    required: true
+  - !ruby/object:Api::Type::String
+    name: "location"
+    description: The location of the resource.
+    immutable: true
+    url_param_only: true
+    required: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: "adminClusterMembership"
+    required: true
+    description: |
+      The Admin Cluster this Bare Metal User Cluster belongs to.
+      This is the full resource name of the Admin Cluster's hub membership.
+    diff_suppress_func: projectNumberDiffSuppress
```

Comment:
This function changed to a capital, I think!

Reply:
> This function changed to a capital, I think!

Thanks for checking. Fixed, but still result in the `Value Conversion Error` (I have rebased on top of main) and followed the [guidance](https://googlecloudplatform.github.io/magic-modules/docs/how-to/add-mmv1-test/#tests-that-use-beta-features) adding `provider = google-beta` and `min_version: beta` accordingly. But still getting the error when running testacc command 

```
make testacc TEST=./google-beta TESTARGS='-run=TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample'
```

Error
```
==> Checking that code complies with gofmt requirements...
go vet
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample
=== PAUSE TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample
=== CONT  TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample
    vcr_utils.go:144: Step 1/2 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid provider configuration
        
        Provider "registry.terraform.io/hashicorp/google-beta" requires explicit
        configuration. Add a provider block to the root module and configure the
        provider's required arguments as described in the provider documentation.
        
        
        Error: Value Conversion Error
        
          with provider["registry.terraform.io/hashicorp/google-beta"],
          on <empty> line 0:
          (source code not available)
        
        An unexpected error was encountered trying to convert tftypes.Value into
        google.ProviderModel. This is always an error in the provider. Please report
        the following to the provider developer:
        
        mismatch between struct and object: Object defines fields not found in
        struct: gkeonprem_custom_endpoint.
--- FAIL: TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample (0.89s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google-beta/google-beta 1.232s
FAIL
make: *** [GNUmakefile:15: testacc] Error 1
```

Is it possibly caused by local environments or anything I could try to locate this issue? @rileykarson @slevenick